[JSON manual] json search function (#61)
CUBRID/cubrid-manual: en/sql/function/json_fn.rst

      json_search('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', 'all', 'a', null, '$**[*]')
    ======================
      "["$.a[0]","$.c[0]","$.d.e[0]"]"

JSON_EXTRACT
===================================

> .. function:: JSON_EXTRACT (json_doc, json path [ , json path] ...)

Reviewer: ```suggestion
.. function:: JSON_EXTRACT (json_doc, json path [, json path] ...)
```

  Returns json elements from the json_doc, that are addressed by the given paths.

>   If json path argments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containg json path are gathered in a resulting json array. 

Reviewer 2: ```suggestion
  If json path arguments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containing json path are gathered in a resulting json array. 
```

>   If json path arguments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containg json path are gathered in a resulting json array. 

Reviewer 2: ```suggestion
  If json path arguments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containing json path are gathered in a resulting json array. 
```

  A single json element is returned if no wildcards are used in the given json paths and a single element is found, otherwise the json elements found are wrapped in a json array.

>   Raises an error if a json path is NULL.  

Reviewer: What happens if a path does not exist?

Author: "Returns NULL if no elements are found"

>   Returns NULL if no elements are found or if json_doc arguments is invalid.

Reviewer: Returns null of json_doc argument is invalid? Shouldn't be an error?

Please note you have a typo `if json_doc arguments is `

.. code-block:: sql

    SELECT JSON_EXTRACT('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', '$.*[*]');
::

      json_extract('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', '$.*[*]')
    ======================
      "["a","b","a"]"

The following json path will match all json paths that end with a json array index (matches all previous matched paths and, in addition, '$.d.e[0]') :

.. code-block:: sql

    SELECT JSON_EXTRACT('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', '$**[*]');
::

      json_extract('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', '$**[*]')
    ======================
      "["a","b","a","a"]"

.. code-block:: sql

    SELECT JSON_EXTRACT('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', '$.d**[*]');
::

      json_extract('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', '$d**[*]')
    ======================
      "["a"]" -- '$.d.e[0]' is the only path matching the given argument path family - paths that start with '.d' and end with an array index

->
===================================

> .. function:: -> (json_doc column, json path)

Reviewer: is there a way to write this as json_doc -> json_path?

  Alias operator for JSON_EXTRACT with 2 arguments and the json_doc argument being a column.

>   Raises an error if a json path is NULL.

Reviewer: ```suggestion
  Raises an error if a json path is NULL or invalid.
```

  Returns NULL if it applied on a NULL json_doc argument.

.. code-block:: sql

    CREATE TABLE tj (a json);

>     INSERT INTO tj values ('{"a":1}'), ('{"a":2}'), ('{"a":2}'), (NULL);

Reviewer 2: You want `3` not redundant `2`.

    SELECT a->'$.a' from tj;
::

      json_extract(a, '$.a')
    ======================
      1
      2
      3
      NULL

->>
===================================

.. function:: ->> (json_doc column, json path)

  Alias for JSON_UNQUOTE(column->json path). Operator can be applied only on json_doc arguments that are columns.

>   Raises an error if a json path is NULL.

Reviewer: ```suggestion
  Raises an error if the json path is NULL or invalid.
```

  Returns NULL if it applied on a NULL json_doc argument.

.. code-block:: sql

    CREATE TABLE tj (a json);

>     INSERT INTO tj values ('{"a":1}'), ('{"a":2}'), ('{"a":2}'), (NULL);

Reviewer 2: `3`?
